packages/ibm-products/src/components/Datagrid/types/index.ts

```diff
@@ -194,7 +194,8 @@ export interface DataGridHeader<T extends object = any>
     UseSortByColumnProps<T> {
   className(className: any, arg1: { [x: string]: any }): unknown;
   isAction?: boolean;
-  slug?: any;
+  slug?: any; // To be removed once the support for slug is not available
+  aiLabel?: any;
```

Comment:
```suggestion
  aiLabel?: ReactNode | boolean;
```

Reply:
Thanks @AlexanderMelox. I've updated the type for both `slug` and `aiLabel` to ReactNode. Since we would only be passing `AILabel` component as value for `aiLabel`, I've not added boolean here.

I've to look into a few more things for this issue, Once that is done I'll change the PR to ready for review